pkg/perf/networkhealth/networkhealth.go

```diff
@@ -0,0 +1,112 @@
+package networkhealth
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/threefoldtech/zos/pkg/environment"
+	"github.com/threefoldtech/zos/pkg/perf"
+)
+
+const requestTimeout = 5 * time.Second
+
+type NetworkHealthTask struct{}
+
+type ServiceStatus struct {
+	Url         string `json:"url"`
+	IsReachable bool   `json:"is_reachable"`
+}
+
+var _ perf.Task = (*NetworkHealthTask)(nil)
+
+func NewTask() *NetworkHealthTask {
+	return &NetworkHealthTask{}
+}
+
+func (t *NetworkHealthTask) ID() string {
+	return "network-health"
+}
+
+func (t *NetworkHealthTask) Description() string {
+	return "Network health check runs periodically to check the connection to most of grid services."
+}
+
+func (t *NetworkHealthTask) Cron() string {
+	return "0 */5 * * * *"
+}
+
+func (t *NetworkHealthTask) Jitter() uint32 {
+	return 0
+}
+
+func (t *NetworkHealthTask) Run(ctx context.Context) (interface{}, error) {
+	env := environment.MustGet()
+	servicesUrl := []string{
+		env.ActivationURL, env.GraphQL, env.FlistURL,
+	}
+	servicesUrl = append(servicesUrl, env.SubstrateURL...)
+	servicesUrl = append(servicesUrl, env.RelayURL...)
+
+	reports := []ServiceStatus{}
+
+	var wg sync.WaitGroup
+	for _, serviceUrl := range servicesUrl {
+		wg.Add(1)
+		go func(serviceUrl string) {
+			defer wg.Done()
+			report := getNetworkReport(ctx, serviceUrl)
+			reports = append(reports, report)
+		}(serviceUrl)
+	}
+	wg.Wait()
+
+	return reports, nil
+}
+
+func getNetworkReport(ctx context.Context, serviceUrl string) ServiceStatus {
+	ctx, cancel := context.WithTimeout(ctx, requestTimeout)
+	defer cancel()
+
+	report := ServiceStatus{
+		Url:         serviceUrl,
+		IsReachable: true,
+	}
+
+	parsedUrl := parseUrl(serviceUrl)
+	err := isReachable(ctx, parsedUrl)
+	if err != nil {
+		report.IsReachable = false
+	}
+
+	return report
+}
+
+func parseUrl(serviceUrl string) string {
+	u, err := url.Parse(serviceUrl)
+	if err != nil {
+		return ""
+	}
+	host := u.Host
+
+	if !strings.Contains(host, ":") {
+		host = fmt.Sprintf("%s:80", host)
+	}
```

Comment:
The default port should be based on the url `scheme` 80 if it's http, and 443 if it's https

Reply:
this was just a workaround for the validation on `net.Dialer.DialContext`.
the services url we call are a mix of different schemes (https/wss/redis) but when i directly use the address on the dial it fails `dial tcp: address tfchain.grid.tf: missing port in address`.
so i am just adding the port 80 for all the address

Reply:
that's what i am saying. Dial because it's a tcp dial needs to know an ip and port. You can't assume that ALL services listening on port 80.  because a single IP can has multiple ports listening (multiple services).

For example say the hub. The hub has a web page that is available on port 80 and 443 (http and https) but also has zdb on port 9900. So if u assume port 80 for all your tests (if the port is not in the host name) then u will only be testing the http port.

Instead this function has to use the default port of the services. 

Luckily links with custom ports already have the port in the URL but your code has to be correct, and to be correct you need to use the default port **of the protocol** when guessing the port number